### ISSUE TITLE
Add doc entries for POD Declarator Blocks

### ISSUE THREAD

**Issue by WildYorkies** [perl6/doc#1230]
# Summary

Per [S26](https://design.perl6.org/S26.html#Declarator_blocks), declarator blocks are a way to attach POD blocks to code objects. 

# Code examples

As seen below, Rakudo properly builds the `Pod::Block::Declarator` object. Yet there is no entry in the docs for this type.

```perl6
my &greet = sub { say "hi"; } #= says hi, ya know?
say &greet.WHY.perl;
# OUTPUT: Pod::Block::Declarator.new(WHEREFORE => sub () { #`(Sub|66820360) ... }, config => {}, contents => [])

#| says hi, ya know?
my &greet = sub { say "hi"; } 
say &greet.WHY.perl;
# OUTPUT is the same as above
```

Rakudo also properly parses the leading and trailing declarators. Yet there is no entry in the docs for `#|` or `#=`.
```perl6
#| says hi, ya know?
my &greet = sub { say "hi"; } #= it does nothing else.
say &greet.WHY;
# OUTPUT: says hi, ya know?
#         it does nothing else.

say &greet.WHY.leading ~ " " ~ &greet.WHY.trailing;
# OUTPUT: says hi, ya know? it does nothing else.

say &greet.WHY.^attributes;
# OUTPUT: (Mu $!WHEREFORE Positional @!leading Positional @!trailing Associative %!config Positional @!contents)

```

Rakudo also works with `.WHEREFORE` called on the declarator object.

```perl6
sub greet() {
    #= it does some stuff
    #= yeah, no joke
    
    say 'stuff'
    
}
say &greet.WHY;
# OUTPUT: it does some stuff yeah, no joke

say &greet.WHY.WHEREFORE;
# OUTPUT: sub greet () { #`(Sub|57757656) ... }
```

# Changes needed

Doc explanations should be added for

- [x] `#|` (Leading declarator)
- [x] `#=` (Trailing declarator)
- [x] `Pod::Block::Declarator` object
- [x] Add entry for `.WHEREFORE` method on `Pod::Block`s
- [x] Update signature for `.WHY` to show that it returns a Declarator object.

**Comment by hankache** [comment]
Fixed by 006efcf 7a6cf0f 72d9d1b

**Comment by hankache** [comment]
@WildYorkies kindly close this issue if this is what you were looking for.
